Userland/Libraries/LibSQL/AST/AST.h

```diff
@@ -6,13 +6,10 @@
 
 #pragma once
 
-#include <AK/NonnullRefPtr.h>
-#include <AK/NonnullRefPtrVector.h>
-#include <AK/RefCounted.h>
-#include <AK/RefPtr.h>
-#include <AK/String.h>
```

Comment:
Yeah it's sometimes hard to figure out what needs to stay and what not. I had an issue caused by a cyclical include of AST.h and one of my headers and I was confused by the error message (imagine that. GCC giving convoluted error messages), so I started stripping out and re-adding headers until it compiled. As it turns out, the ones above are not necessary to make it compile. But I guess policy is to include headers of types that are in the source file?

Reply:
Oh I see the problem now. Those headers were probably "accidentally" already included by all files including AST.h.

Reply:
Right - the general rule in C++ is that each file should include the headers for whatever it directly uses, rather than relying on other files to include its dependencies on its behalf. So `AST.h` shouldn't rely on files that include `AST.h` to also include, e.g., `String.h` (nor should `AST.h` depend on headers it includes itself to include `String.h`).